File: backend/src/models/TimeLog.js
const mongoose = require("mongoose");
const { Schema } = mongoose;

const timeLogSchema = new Schema(
  {
    cardId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Card",
      required: true,
    },
    startTime: {
      type: Date,
      required: true,
    },
    endTime: {
      type: Date,
      default: null,
    },
    duration: {
      type: Number,
      default: 0,
    },
  },
  {
    timestamps: true,
  },
);

//construct Schema into a model
const TimeLog = mongoose.model("TimeLog", timeLogSchema);

module.exports = TimeLog;

File: backend-dotnet/Models/TimeLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NoBacklog.Api.Models;

public class TimeLog
{
    public int Id { get; set; }

    [Required]
    public int CardId { get; set; }

    [Required]
    public DateTime StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public long Duration { get; set; } = 0;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey(nameof(CardId))]
    public Card Card { get; set; } = null!;
}